tests/integration/smoke/src/test/java/org/apache/bookkeeper/tests/integration/TestBookieShellCluster.java

```diff
@@ -83,4 +93,98 @@ public void test002_ListROBookies() throws Exception {
     public void test003_ListRWBookies() throws Exception {
         super.test003_ListRWBookies();
     }
+
+    /**
+     * These tests on being able to access cluster internals, so can't be put in test base.
+     */
+    @Test
+    public void test101_RegenerateIndex() throws Exception {
+        String zookeeper = String.format("zk+hierarchical://%s/ledgers",
+                                         BookKeeperClusterUtils.zookeeperConnectString(docker));
+        int numEntries = 100;
+
+        long ledgerId = 0;
+
+
+        try (BookKeeper bk = BookKeeper.newBuilder(
+                     new ClientConfiguration().setMetadataServiceUri(zookeeper)).build()) {
+            log.info("Writing entries");
+            try (WriteHandle writer = bk.newCreateLedgerOp().withEnsembleSize(1)
+                    .withWriteQuorumSize(1).withAckQuorumSize(1)
+                    .withPassword("".getBytes()).execute().get()) {
+                int i = 0;
+                for (; i < numEntries - 1; i++) {
+                    writer.appendAsync(("entry" + i).getBytes());
+                }
+                writer.append(("entry" + i).getBytes());
+
+                ledgerId = ledgerId;
+            }
+
+            log.info("Restart bookies so index gets flushed");
+            assertTrue(BookKeeperClusterUtils.stopAllBookies(docker));
+            assertTrue(BookKeeperClusterUtils.startAllBookiesWithVersion(docker, currentVersion));
+
+            log.info("Find the bookie containing the index");
+            String indexFileName = String.format("/opt/bookkeeper/data/ledgers/current/0/0/%d.idx", ledgerId);
+            List<String> bookie = BookKeeperClusterUtils.allBookies().stream().filter(
+                    (b) -> {
+                        try {
+                            DockerUtils.runCommand(docker, b, "test", "-f", indexFileName);
+                            return true;
+                        } catch (Exception e) {
+                            return false;
+                        }
+                    }).collect(Collectors.toList());
+            Assert.assertEquals(bookie.size(), 1); // should only be on one bookie
+
+            log.info("Validate that we can read back");
+            try (ReadHandle reader = bk.newOpenLedgerOp()
+                    .withLedgerId(ledgerId)
+                    .withPassword("".getBytes()).execute().get();
+                 LedgerEntries entries = reader.read(0, numEntries - 1)) {
+                Assert.assertEquals(reader.getLastAddConfirmed(), numEntries - 1);
+
+                for (int i = 0; i < numEntries; i++) {
+                    Assert.assertEquals(new String(entries.getEntry(i).getEntryBytes()), "entry" + i);
```

Comment:
add UTF_8; put expected value as the first param and actual value as the second param

Reply:
>  It's super annoying how junit and testng have different orders.

yeah, I felt annoyed as well